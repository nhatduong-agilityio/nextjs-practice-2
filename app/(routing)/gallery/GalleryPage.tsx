'use client'

import { Flex, Grid, GridItem, Heading, Stack } from '@chakra-ui/react'
import React, { Fragment, memo } from 'react'
import NextLink from 'next/link'
import Image from '@components/Common/Image'
import { switcherGridItemProps } from '@helper/gridGallery'
import type { TArticles } from '@matched-types/articles'
import { PAGE_URL } from '@constants/routes'

interface IProps {
  articles: TArticles
}

const GalleryPage = memo(({ articles }: IProps) => (
  <Flex w='full' flexDir='column' p={['10px 20px', '50px 100px', '50px 100px']} pt='30px'>
    <Stack
      flexDir='row'
      justifyContent='flex-start'
      alignItems='center'
      pb={['25px', '50px', '50px']}
    >
      <Heading variant={['extraBoldMD', 'extraBoldXL']} lineHeight='inherit'>
        Interior Designer
      </Heading>
    </Stack>
    <Grid
      h='100vh'
      templateRows={['repeat(5, 1fr)', 'repeat(7, 1fr)']}
      templateColumns={['repeat(4, 1fr)']}
      gap={['10px', '50px', '50px']}
    >
      {articles.map((item, index) => (
        <Fragment key={item.id}>
          {/** function helper render gird item props by index of array */}
          <GridItem {...switcherGridItemProps(index)}>
            <NextLink href={`${PAGE_URL.GALLERY.URL}/${item.id}`} passHref>
              <Image alt={item.imageAlt} src={item.image} width='full' height='full' />
            </NextLink>
          </GridItem>
        </Fragment>
      ))}
    </Grid>
  </Flex>
))

GalleryPage.displayName = 'GalleryPage'

export default GalleryPage
