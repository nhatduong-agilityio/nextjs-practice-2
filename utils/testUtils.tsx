import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
