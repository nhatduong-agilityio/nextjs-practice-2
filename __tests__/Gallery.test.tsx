import type { RenderResult } from '@utils/testUtils'
import { cleanup, render, waitFor } from '@utils/testUtils'
import { ArticlesMock } from '@mock/dataMock'
import Gallery from '../app/(routing)/gallery/page'
import GalleryPage from '../app/(routing)/gallery/GalleryPage'

jest.mock('next/navigation', () => ({
  notFound: jest.fn(),
  useSearchParams: jest.fn(),
}))

jest.mock('@services/requests', () => ({
  fetcherInstance: jest.fn(),
}))

jest.doMock('../app/(routing)/gallery/GalleryPage.tsx', () => {
  const GalleryPageMock = () => <GalleryPage articles={ArticlesMock} />
  return GalleryPageMock
})

describe('Gallery', () => {
  let wrapper: RenderResult

  const setup = async () => {
    const element = await Gallery()

    const view = render(element)

    await waitFor(() => {
      wrapper = view
    })
  }

  beforeEach(() => {
    cleanup()

    setup()
  })

  it('Should Gallery match snapshot', () => {
    const { container } = wrapper

    expect(container).toMatchSnapshot()
  })
})
