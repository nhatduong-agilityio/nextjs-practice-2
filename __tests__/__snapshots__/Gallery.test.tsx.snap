// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Should Gallery match snapshot 1`] = `
<div>
  <div
    class="css-1gw30fa"
  >
    <div
      class="chakra-stack css-1yykub2"
    >
      <h2
        class="chakra-heading css-1k4oj01"
      >
        Interior Designer
      </h2>
    </div>
    <div
      class="css-121qhnu"
    >
      <div
        class="css-1lz6fgg"
      >
        <a
          href="/gallery/794b844d-d446-4146-a46f-1ab4026cef7c"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-2bei10"
      >
        <a
          href="/gallery/4cf37407-6fc0-4b05-99ca-df755ab5546a"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-15iqidc"
      >
        <a
          href="/gallery/9b320e17-8e4a-4a31-9df0-4b8e994290e5"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-nrxmu7"
      >
        <a
          href="/gallery/05e112cf-5cd7-4f39-94f4-c037e8c107e5"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-1xz837u"
      >
        <a
          href="/gallery/19433980-b25f-451a-94fe-2aad166ea563"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-plg8k9"
      >
        <a
          href="/gallery/f2ccf7e5-5723-4298-a9e5-c5213f6b37c1"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa2"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-4gox30"
      >
        <a
          href="/gallery/9e155b22-df51-4b2c-b91d-07a9ac68678b"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-18i2y6i"
      >
        <a
          href="/gallery/d04939c7-522a-4690-baed-4235e29835ca"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-ku5lu7"
      >
        <a
          href="/gallery/a2078a47-dc28-4fdf-a0b5-1e01d7b0852c"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa3"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-hfwe92"
      >
        <a
          href="/gallery/2fffc81d-747e-481e-9eb6-926d77322565"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture chair and table office4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-hfwe92"
      >
        <a
          href="/gallery/16c27b1d-8cde-41bd-8a89-2419aa9bb28f"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture desk office4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
      <div
        class="css-1803f6"
      >
        <a
          href="/gallery/2522be73-ee1d-4f5d-a216-8dd04d5da3b4"
        >
          <div
            class="css-ackq5e"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Picture of furniture sofa4"
                data-nimg="fill"
                decoding="async"
                sizes="100vw"
                src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
